datafusion/core/src/datasource/physical_plan/parquet/opener.rs

```diff
@@ -126,6 +126,7 @@ impl FileOpener for ParquetOpener {
             let options = ArrowReaderOptions::new().with_page_index(enable_page_index);
 
             let mut metadata_timer = file_metrics.metadata_load_time.timer();
+            // The parquet open action happens here, first the metadata
```

Comment:
@rok This is where the magic happens to link datafusion to parquet. Basically, we need links for the metadata load_ansyc and then the reader.